kubernetes/controller/internal/controller/component/component_controller.go

```diff
@@ -248,11 +262,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 	}
 
 	desc, err := cacheBackedRepo.GetComponentVersion(ctx, component.Spec.Component, version)
-	// TODO: Adjust when https://github.com/open-component-model/ocm-project/issues/784 is merged (@frewilhelm @
 	if errors.Is(err, workerpool.ErrResolutionInProgress) {
+		// Resolution is in progress, the controller will be re-triggered via event source when resolution completes
 		status.MarkNotReady(r.EventRecorder, component, v1alpha1.ResolutionInProgress, err.Error())
+		logger.Info("component version resolution in progress, waiting for event notification",
+			"component", component.Spec.Component,
+			"version", version)
 
-		return ctrl.Result{}, fmt.Errorf("failed to get component version: %w", err)
+		// We requeue as a fallback. This is to avoid stuck component versions if the event queue was full.
+		// Naturally, this component version should be notified way before this timeout.
+		return ctrl.Result{RequeueAfter: fallbackRecoveryTimeout}, nil
```

Comment:
imho this should not be done. we have the controller resync interval for this which is configurable by default and set to 1 hour currently I believe. WDYT?

if you wanna keep this imho it should be set to the sync period of the cr

Reply:
Fair. I was just thinking of defensiveness.